### PR TITLE
Hide login when client id and secret are not set

### PR DESCRIPTION
**Pull request by shun-fan**
I also moved setting the bucket attribute with setting the auth check

@tswast

### CODE REVIEW

File: bookshelf/src/main/webapp/base.jsp [hunk 1]
```diff
@@ -34,9 +34,11 @@ Copyright 2015 Google Inc. All Rights Reserved.
         </div>
         <ul class="nav navbar-nav">
           <li><a href="/">Books</a></li>
-          <li><a href="/books/mine">My Books</a></li>
+          <div class="form-group ${isAuthConfigured ? '' : 'hidden'}">
```
Comment (tswast) [inline bookshelf/src/main/webapp/base.jsp]: I did `<c:if test="${isOAuth2Configured}"><li><a href="/books/mine">My Books</a></li></c:if>` in my version (still pending because I'm going through the rest of the tutorial so I can test it. I think `c:if` would be a little more clear than a hidden div.

File: bookshelf/src/main/webapp/base.jsp [hunk 2]
```diff
@@ -51,7 +53,7 @@ Copyright 2015 Google Inc. All Rights Reserved.
           <a href="/login">Login</a>
```
Comment (tswast) [inline bookshelf/src/main/webapp/base.jsp]: I changed the `c:otherwise` into `<c:when test="${isOAuth2Configured}">`. I wouldn't expect the token to be populated when auth is not configured, so this made sense to me.

Otherwise, I think we can use a `c:if` over the how navbar, since these elements are not needed in any form inputs, so they can be not present rather than hidden with CSS.

File: bookshelf/src/main/java/com/example/managedvms/gettingstartedjava/basicactions/CreateBookServlet.java [hunk 3]
```diff
@@ -46,9 +44,6 @@
   @Override
   public void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException,
       IOException {
-    req.setAttribute(
```
Comment (tswast) [inline bookshelf/src/main/java/com/example/managedvms/gettingstartedjava/basicactions/CreateBookServlet.java]: Why is this moved out of the Create view?

Comment (shun-fan) [inline bookshelf/src/main/java/com/example/managedvms/gettingstartedjava/basicactions/CreateBookServlet.java]: I moved it so that session variables could be set in a central location. Also, since the setting doesn't change throughout the life of the app, I thought it made sense to set it once instead of checking with every new request

File: bookshelf/src/main/webapp/base.jsp (hunk 1 above)
Comment (shun-fan) [inline bookshelf/src/main/webapp/base.jsp]: Ok, this and the next comment make sense. If you want to continue testing your version we can use that. My only suggestion would be do all the property checking in one place, such as the init function in ListBookServlet

Comment (tswast) [inline bookshelf/src/main/webapp/base.jsp]: I'm not sure I understand.

>>  do all the property checking in one place, such as the init function in ListBookServlet

How would the requests to the CreateServlet get populated by the ListBookServlet code? I guess I could refactor some of this into a "populateConfiguration" method or something.

Comment (tswast) [inline bookshelf/src/main/webapp/base.jsp]: Oh. I see now. JSPs can read from the servlet context.

Comment (tswast) [inline bookshelf/src/main/webapp/base.jsp]: Oh, but we aren't guaranteed that ListServlet will be the first one called. I think I will make a helper utility and make sure it's called from each.

**Comment by tswast** [comment]
I've merged these changes with https://github.com/GoogleCloudPlatform/getting-started-java/pull/65.

I am having trouble testing locally. Login makes it through flow, but then I'm not logged in in the end. Works when deployed, though.

File: bookshelf/src/main/webapp/base.jsp (hunk 1 above)
Comment (shun-fan) [inline bookshelf/src/main/webapp/base.jsp]: actually the webservlet annotation has a loadonstartup = 1. this guarantees that the servlet is loaded first.
